File: GbrainsJavaProject/app/src/main/java/com/nbgsoftware/gbrains/ui/signup/SignUpFragment.java
package com.nbgsoftware.gbrains.ui.signup;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.nbgsoftware.gbrains.baseMVP.MVPFragmentPresenter;
import com.nbgsoftware.gbrains.databinding.FragmentSignUpBinding;

public class SignUpFragment extends MVPFragmentPresenter<
        FragmentSignUpBinding,
        SignUpContract.View,
        SignUpContract.Presenter<SignUpContract.View>>
        implements SignUpContract.View {
    @Override
    public FragmentSignUpBinding getViewBinding(LayoutInflater inflater, ViewGroup container) {
        return FragmentSignUpBinding.inflate(inflater, container, false);
    }

    @Override
    public SignUpContract.Presenter<SignUpContract.View> getPresenter() {
        return new SignUpPresenter<>();
    }

    @Override
    public void initialize() {

    }
}

File: InstagramClone/app/src/main/java/com/instagram/features/sign_up/SignUpFragment.kt
package com.instagram.features.sign_up

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.lifecycle.ViewModelProvider
import com.google.android.material.tabs.TabLayoutMediator
import com.instagram.base.BaseFragmentViewModel
import com.instagram.databinding.FragmentSignUpBinding

class SignUpFragment : BaseFragmentViewModel<FragmentSignUpBinding, SignUpViewModel>() {

    override fun createViewBinding(
        inflater: LayoutInflater,
        container: ViewGroup?
    ): FragmentSignUpBinding {
        return FragmentSignUpBinding.inflate(inflater, container, false)
    }

    override fun createViewModel(): SignUpViewModel {
        return ViewModelProvider(this)[SignUpViewModel::class.java]
    }

    override fun bindingStateView() {
        super.bindingStateView()
        val registerPagerAdapter = RegisterPagerAdapter(requireActivity())
        this.binding.viewPager.adapter = registerPagerAdapter
        TabLayoutMediator(
            this.binding.tabLayout,
            this.binding.viewPager,
            TabLayoutMediator.TabConfigurationStrategy { tab, position ->
                when (position) {
                    0 -> tab.text = "Phone"
                    1 -> tab.text = "Email"
                }
            }).attach()
    }
}